File: js/emit_numeric_exponent.js
function
emit_numeric_exponent(u, p) // p is rational or double, sign is not emitted
{
	var v = {type:SUPERSCRIPT, a:[], small_font:1};

	if (isrational(p)) {
		emit_roman_text(v, Math.abs(p.a).toFixed(0));
		if (p.b != 1) {
			emit_roman_text(v, "/");
			emit_roman_text(v, p.b.toFixed(0));
		}
	} else
		emit_double(v, p);

	emit_update_superscript(u, v);

	u.a.push(v);
}

File: app/emit_numeric_exponent.c
#include "app.h"

// p is rational or double, sign is not emitted

void
emit_numeric_exponent(struct atom *p)
{
	int t;
	char *s;

	emit_level++;

	t = tos;

	if (isrational(p)) {
		s = mstr(p->u.q.a);
		emit_roman_string(s);
		if (!MEQUAL(p->u.q.b, 1)) {
			emit_roman_char('/');
			s = mstr(p->u.q.b);
			emit_roman_string(s);
		}
	} else
		emit_double(p);

	emit_update_list(t);

	emit_level--;

	emit_update_superscript();
}
